File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand1, strand2)
    new(strand1, strand2).distance
  end

  def initialize(strand1, strand2)
    return raise ArgumentError, 'Strand lengths must be of identical length' unless strand1.length == strand2.length

    @nucleotides = strand1.chars.zip(strand2.chars)
  end

  def distance
    @nucleotides.count { |c| c.first != c.last }
  end
end

File: kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        if (leftStrand.length != rightStrand.length) {
            throw IllegalArgumentException("left and right strands must be of equal length")
        }
        var counter = 0
        leftStrand.zip(rightStrand) {l,r -> if(l != r) counter ++}
        return counter
    }
}
